### PR TITLE
Supporting PSR-17 and PSR-18

### PR DESCRIPTION
**Pull request by gmponos**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -47,5 +47,7 @@
     "suggest": {
         "php-http/message": "Required to use Guzzle for sending HTTP requests",
         "php-http/guzzle6-adapter": "Required to use Guzzle for sending HTTP requests"
-    }
+    },
+    "prefer-stable": true,
+    "minimum-stability": "dev"
```
Comment (gmponos) [inline composer.json]: Fix this once php-http is ready.

**Comment by florianv** [comment]
@gmponos The issue is that many people use Swap with Geocoder (got a lot of issues related to the usage of both) and right now Geocoder only supports `httplug` version 1.0, so I don't think we can use 2.0 unfortunately.

@Nyholm Are there plans to support `httplug 2.0` in `Geocoder` ?

**Comment by gmponos** [comment]
maybe then this PR and this plan is premature.. I guess we have to wait for @php-http to finalize it's work about PSR-17/18.

**Comment by gmponos** [comment]
Doesn't this cover us?

https://github.com/geocoder-php/Geocoder/blob/master/src/Http/composer.json#L17

**Comment by Nyholm** [comment]
True @gmponos, that is enough. 

About the main package. That is just a monolith repo. There is actually no Packagist package for that repo. But sure, the line you are referring to should be updated.

**Comment by florianv** [comment]
@gmponos Are you willing to send more updates or should I closed it ?

**Comment by gmponos** [comment]
Hello @florianv .. I made the changes.. 

Sorry for the delay.. some things were needed to be fixed on PHP-HTTP packages...

Maybe this PR could be merged and fix composer later... your choice :)

ping @Nyholm if he has time to have a look as well..

**Comment by gmponos** [comment]
Argg.. need to fix prefer lowest test... wont take long...

**Comment by florianv** [comment]
@gmponos I pushed more stuff, so it will have to be rebased!

We can wait for `httplug` to be ready, no problem.

**Comment by Nyholm** [comment]
We just updated php-http/discovery:dev-master

**Comment by florianv** [comment]
@gmponos Do you think the PR is ready ? Looks good to me!

**Comment by gmponos** [comment]
Yes it could be merged.. but before tagging the current package composer must be fixed.

**Comment by florianv** [comment]
Hey @gmponos, it looks like 1.6 as released!

**Pull request merged**